python/ray/data/dataset_iterator.py

```diff
@@ -700,3 +704,11 @@ def _is_tensor_dataset(self) -> bool:
         if schema is None or isinstance(schema, type):
             return False
         return _is_tensor_schema(schema.names)
+
+    def _best_batch_format(self) -> Literal["default", "pandas", "pyarrow", "numpy"]:
+        """Returns the best batch format that lines up with the dataset format.
```

Comment:
What does it mean "best"? Matching the dataset format doesn't necessarily has the best perf, e.g. pyarrow-to-pandas conversion can be faster than pandas-to-pandas conversion.

Reply:
@jianoaix - I am open to a better name, do you have a suggestion for naming?

Reply:
`_default_batch_format` to align with Dataset (the idea of "best" can be in comment)?
Also since here is a best determined batch format, should it also be used by `iter_batches` when the batch_format arg is not specified by users?

Reply:
I guess I won't touch the batch format behavior for `iter_batches`, that would trigger more change and dsicussion across `Dataset` and `DatasetIterator`. ideally I want this PR should be simple one to unblock CI test failure. I can do the renaming part.

Reply:
It should be simple but this invites a question why do we have a class member which determines the best format in the context of iterator, but is not actually used for iter_batches().

Reply:
The current behavior for `iter_batches` is always using the "default" if user does not provide. The `iter_rows` behavior is different - in bulk execution, it checks `Dataset.dataset_format()` and uses the returned one, while in streaming execution, it uses "default".

The difference is already happening before this PR, and here we just introduce a private method in `DatasetIterator`, so it should be easy to change/remove later. Can we defer the behavior change for `iter_batches` to a future PR?

Reply:
Yeah, SG